torch/_dynamo/polyfills/__init__.py

```diff
@@ -101,6 +101,10 @@ def accumulate_grad(x, new_grad):
         x.grad.add_(new_grad_strided)
 
 
+def is_iterable(obj):
```

Comment:
Use a typing_extensions.TypeIs[Iterable] as a return type to propagate the type narrowing.